AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -345,12 +367,67 @@ - (void)createOrReplaceWithPartition:(NSString *)partition
                    }];
 }
 
+- (void)replaceWithPartition:(NSString *)partition
+                  documentId:(NSString *)documentId
+                    document:(id<MSSerializableDocument>)document
+                writeOptions:(MSWriteOptions *)writeOptions
+           completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+  
+  // In the current version we do not support E-tag optimistic concurrency logic and replace will call create.
+  [[MSDataStore sharedInstance] createOrReplaceWithPartition:partition
+                                                  documentId:documentId
+                                                    document:document
+                                                writeOptions:writeOptions
+                                           additionalHeaders:@{kMSDocumentUpsertHeaderKey : @"true"}
+                                           completionHandler:completionHandler];
+}
+
+- (void)deleteDocumentWithPartition:(NSString *)partition
+                         documentId:(NSString *)documentId
+                       writeOptions:(MSWriteOptions *)__unused writeOptions
+                  completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
+  [self performOperationForPartition:partition
+                          documentId:documentId
+                          httpMethod:kMSHttpMethodDelete
+                                body:[NSData data]
+                   additionalHeaders:nil
+                   completionHandler:^(NSData *__unused data, NSError *_Nonnull cosmosDbError) {
+                     // Body returned from call (data) is empty.
+                     NSInteger httpStatusCode = [MSDataSourceError errorCodeFromError:cosmosDbError];
+                     if (httpStatusCode != MSHTTPCodesNo204NoContent) {
+                       MSLogError([MSDataStore logTag],
+                                  @"Not able to delete document. Error: %@; HTTP status code: %ld; "
+                                  @"Document: %@/%@",
+                                  cosmosDbError.localizedDescription, (long)httpStatusCode, partition, documentId);
+                     } else {
+                       MSLogDebug([MSDataStore logTag], @"Document deleted: %@/%@", partition, documentId);
+                     }
+                     completionHandler([[MSDataSourceError alloc] initWithError:cosmosDbError]);
```

Comment:
this looks broken, why does it return an error when delete is successful?

Reply:
We moved it only. Out of scope.